libs/server/src/lib/utils/read-generator-collections.ts

```diff
@@ -155,29 +158,42 @@ async function readCollection(
       packageJson.json.schematics || packageJson.json.generators,
       dirname(packageJson.path)
     );
-    return readCollectionSchematics(collectionName, collection.json);
+    return readCollectionGenerators(collectionName, collection.json);
   } catch (e) {
     // this happens when package is misconfigured. We decided to ignore such a case.
     return null;
   }
 }
 
-async function readCollectionSchematics(
+function readCollectionGenerators(
   collectionName: string,
   collectionJson: any
-) {
-  const generators = new Set<Schematic>();
+): GeneratorCollection {
+  const generators = new Set<Generator>();
 
   try {
     Object.entries(
       Object.assign({}, collectionJson.schematics, collectionJson.generators)
-    ).forEach(async ([k, v]: [any, any]) => {
+    ).forEach(([k, v]: [any, any]) => {
       try {
         if (canAdd(k, v)) {
+          let generatorType: GeneratorType;
```

Comment:
Should we include case's for 'app' & 'lib' here as well? This is not something that we currently standardize when thinking about community plugins. As a real example, my plugin nx-dotnet currently uses app / lib instead of application and library.